src/coins.cpp

```diff
@@ -76,13 +76,13 @@ void CCoinsViewCache::AddCoin(const COutPoint &outpoint, Coin&& coin, bool possi
     bool inserted;
     std::tie(it, inserted) = cacheCoins.emplace(std::piecewise_construct, std::forward_as_tuple(outpoint), std::tuple<>());
     bool fresh = false;
+    if (!possible_overwrite && !it->second.coin.IsSpent()) { // check before any other partial state change
+        throw std::logic_error("Attempted to overwrite an unspent coin (when possible_overwrite is false)");
+    }
     if (!inserted) {
```

Comment:
Would it be simpler to just move this `if` block below the `if` block that throws, instead of pulling the throw out?

Reply:
If you think that's better, I don't mind - done